test/fuzzer/pedro/gcol_stack_overflow.test

```diff
@@ -0,0 +1,16 @@
+# name: test/fuzzer/pedro/gcol_stack_overflow.test
+# group: [pedro]
+
+statement ok
+pragma enable_verification
+
+statement ok
+CREATE TABLE t0 (c2 AS (c0) VIRTUAL, c0 INT);
+
+query I
+SELECT 1 FROM t0 t2(c0) WHERE c0 = 0;
```

Comment:
Does this also work if we alias the table to its own name (e.g. t0 AS t0(c0))

Reply:
Will add more tests for that - but it should, as I already had a step to bake the table name into the expression during binding